diff_cover/util.py

```diff
@@ -1,5 +1,35 @@
+import contextlib
 import os.path
 import posixpath
+import sys
+
+
+@contextlib.contextmanager
+def open_file(path, mode, encoding="utf-8"):
+    """
+    Behaves like open(), but with some special cases for stdout and stderr.
+
+    :param path: string of the path to open
+    :param mode: string of the mode to open the file in
+    :param encoding: encoding to use when opening the file (text mode only)
+    :return: a context manager that yields the file object
+    """
+    if path in ("/dev/stdout", "-"):
+        output_file = sys.stdout
+    elif path == "/dev/stderr":
+        output_file = sys.stderr
+    else:
+        if "b" in mode:
+            encoding = None
+
+        with open(path, mode, encoding=encoding) as f:
+            yield f
+        return
+
+    if "b" in mode:
+        output_file = output_file.buffer
+
+    yield output_file
```

Comment:
Mostly a style tweak, if you feel passionately whats already here then whatever. I just figuring out the variables up top and then deciding behavior towards the bottom 

```
   output_file = None
   if path in ("/dev/stdout", "-"):
       output_file = sys.stdout
   elif path == "/dev/stderr":
       output_file = sys.stderr
   elif "b" in mode:
       output_file = output_file.buffer
    
   if output_file:
       yield output_file
   else:
       if "b" in mode:
           encoding = None

       with open(path, mode, encoding=encoding) as f:
           yield f
   ```